skypy/halo/abundance_matching.py

```diff
@@ -14,36 +14,75 @@
 
 """
 
-from astropy.table import hstack
+import numpy as np
+from skypy.halo.mass import press_schechter, number_subhalos, subhalo_mass_sampler
+from skypy.galaxy.luminosity import schechter_lf_magnitude
 
+__all__ = [
+    'vale_ostriker',
+]
 
-def vale_ostriker(halos, galaxies, mass='mass', luminosity='luminosity',
-                  join_type='inner'):
+
+def vale_ostriker(halo_kwargs, subhalo_kwargs, galaxy_kwargs):
     """Vale & Ostriker abundance matching.
-    Takes catalogs of halos and galaxies and performs abundance matching
-    following the method outlined in Vale & Ostriker (2004) assuming
-    monotonicity between halo mass and galaxy luminosity to return an
-    abundance-matched table of halo-galaxy pairs.
+    Generate matched arrays of (sub)halos masses and galaxy absolute magnitudes
+    following the abundance matching model in Vale & Ostriker (2004).
 
     Parameters
     ----------
-    halos, galaxies : Astropy Table
-        Tables of halos and galaxies to be matched.
-    mass, luminosity : str
-        Halo mass and galaxy luminosity column names.
-    join_type : str
-        Join type (‘inner’ | ‘exact’ | ‘outer’), default is inner
+    halo_kwargs : dict
+        Dictionary of keyword arguments for skypy.halo.press_schechter.
+    subhalo_kwargs : dict
+        Dictionary of keyword arguments for skypy.halo.number_subhalos and
+        skypy.halo.subhalo_mass_sampler.
+    galaxy_kwargs : dict
+        Dictionary of keyword arguments for skypy.galaxy.schechter_lf_magnitude.
 
     Returns
     -------
-    matched_Table : Astropy Table
-        Table of abundance-matched halos and galaxies.
-
+    mass : array_like
+        Array of (sub)halo masses.
+    group : array_like
+        Array of halo group ids.
```

Comment:
```suggestion
        Array of halo group identity number.
```
Avoid abbreviations.

Reply:
oh come on